Sources/NIO/Codec.swift

```diff
@@ -429,3 +436,13 @@ extension MessageToByteEncoder {
         return ctx.channel.allocator.buffer(capacity: 256)
     }
 }
+
+extension ByteToMessageHandler: RemovableChannelHandler {
+    public func removeHandler(ctx: ChannelHandlerContext, promise: EventLoopPromise<Void>?) {
+        ctx.eventLoop.execute {
+            self.processLeftovers(ctx: ctx)
+            assert(self.state != .leftoversNeedProcessing)
+            ctx.removeHandler(promise: promise)
+        }
```

Comment:
This implementation is a bit awkward: specifically, it doesn't stop the decode loop if one is running. This means it's entirely possible for the `B2MD` to process more data. It would probably be nicer if this interrupted the decode loop.